In [30]:
import pandas as pd
import statistics as st
import math
from scipy.stats import norm
import numpy as np

# TASK 1

GIVEN:-

    sd=1.28
    
    HYPOTHESIS
    
    NULL HYPOTHESIS(H0)
    (i.e)u1-u2=0
    
    ALTERNATE HYPOTHESIS(Ha)
    (i.e)u1-u2!=0
    
    n=49
    
    SL=5%
    

In [26]:
#They are two tailed test
s_mean=32.18
p_mean=33.88  
p_sd=1.28
n=49
alpha=0.05  

Z-Statistics= -9.296875000000016
P-Value= 1.4463410868771357e-20
Z-Critical value= -1.9599639845400545


In [70]:
#Analysing data
z_statistic=(s_mean-p_mean)/(p_sd/math.sqrt(n))
print('Z-Statistics=',z_statistic)

p_value=norm.sf(abs(z_statistic))*2  
print('P-Value=',p_value)

z_critical=norm.ppf(alpha/2)     
print('Z-Critical value=',z_critical)


Z-Statistics= -9.296875000000016
P-Value= 1.4463410868771357e-20
Z-Critical value= -2.575829303548901


In [68]:
#Reaching statistical conclusion
print(p_value<alpha)
print(z_critical>z_statistic)   #  here z-critical is negative

True
True


In [63]:
#MAKE A BUSINESS DECISION


From our statistical conclusion,we confirm the rejection of Null hypothesis(H0)

So,Alternate hypothesis(Ha) is considered where (u1-u2)!=0  (i.e)difference between the population and sample mean is not zero

# TASK 2

Alternate hypothesis(Ha) >85

Null hypothesis(H0)  <=85


In [72]:
#One side RIGHT tailed test
p_mean=85
p_sd=2.1
n=32
s_mean=87.5
alpha=0.05

In [74]:
#Analysing data
z_statistic=(s_mean-p_mean)/(p_sd/math.sqrt(n))
print('Z-Statistics=',z_statistic)

p_value=norm.sf(abs(z_statistic))  
print('P-Value=',p_value)

z_critical=norm.ppf(1-alpha)     
print('Z-Critical value=',z_critical)

Z-Statistics= 6.734350297014738
P-Value= 8.233201952582752e-12
Z-Critical value= 1.6448536269514722


In [78]:
#Reaching statistical conclusion
print(p_value<alpha)
print(z_statistic>z_critical)

True
True


In [79]:
#MAKE A BUSINESS DECISION

From our statistical conclusion,we confirm the rejection of Null hypothesis(H0)

So,Alternate hypothesis(Ha) is considered (i.e)The different grip helped Mac throw farther than 85 yards 

# TASK 3

TO DETERMINE THE DIFFERENCE IN AVG PRICE OF A LITER OF SPECIFIC HEALTH DRINK IN CALIFORNIA AND FLORIDA.

California-->33 sample

Florida-->30

    NULL HYPOTHESIS(H0)
    (i.e)u1-u2=0
    
    ALTERNATE HYPOTHESIS(Ha)
    (i.e)u1-u2!=0

In [6]:
#GIVEN:-
var_cal=0.028
var_flo=0.015

In [7]:
pwd

'C:\\Users\\Admin'

In [8]:
city_df=pd.read_csv('DS1_C5_S6_Hypothesis_I_Practice_CityPrice_Data.csv')
city_df

California  Florida
0         2.55     2.25
1         2.36     2.40
2         2.38     2.19
3         2.43     2.39
4         2.73     2.39
5         2.67     2.50
6         2.54     2.33
7         2.24     2.33
8         2.43     2.40
9         2.80     2.49
10        2.50     2.19
11        2.64     2.29
12        2.25     2.59
13        2.38     2.23
14        2.61     2.41
15        2.71     2.56
16        2.80     2.48
17        2.49     2.29
18        2.69     2.28
19        2.43     2.39
20        2.23     2.59
21        2.61     2.58
22        2.57     2.53
23        2.36     2.26
24        2.57     2.39
25        2.56     2.58
26        2.38     2.29
27        2.43     2.27
28        2.71     2.45
29        2.25     2.56
30        2.64      NaN
31        2.27      NaN
32        2.18      NaN

In [9]:
#CENTRAL TENDENCY

print('SIZE OF CITY DF=',len(city_df),'\n')

print('CALIFORNIA')
n1=len(city_df)
print('SIZE=',n1)
Mean=st.mean(city_df['California'])
print('MEAN=',Mean)
sd1=math.sqrt(var_cal)
print('SD=',sd1,'\n')

c_df=city_df.copy()
c_df=c_df.dropna()

print('FLORIDA')
n2=len(c_df)
print('SIZE=',n2)
Mean1=st.mean(c_df['Florida'])
print('MEAN=',Mean1)
sd2=math.sqrt(var_flo)
print('SD=',sd2)


SIZE OF CITY DF= 33 

CALIFORNIA
SIZE= 33
MEAN= 2.4966666666666666
SD= 0.1673320053068151 

FLORIDA
SIZE= 30
MEAN= 2.396
SD= 0.1224744871391589


CONFIDENCE LEVEL=99%

In [75]:
#They are two tailed test
alpha=0.01

#Analysing data
z_statistics=((Mean-Mean1)/(math.sqrt(var_cal/n1+ var_flo/n2)))
print('Z STATISTICS=',z_statistics)

p_value=norm.sf(abs(z_statistics))*2
print('P-VALUE=',p_value)

z_critical=norm.ppf(1-alpha/2)
print('Z-CRITICAL=',z_critical)

Z STATISTICS= 2.7413384193834984
P-VALUE= 0.006118944821314574
Z-CRITICAL= 2.5758293035489004


In [11]:
#Reaching statistical conclusion
print(p_value<alpha)
print(z_statistics>z_critical)

True
True


In [62]:
#MAKE A BUSINESS DECISION


From our statistical conclusion,we confirm the rejection of Null hypothesis(H0)

So,Alternate hypothesis(Ha) is considered where (u1-u2)!=0  (i.e)difference between the mean price of health drink between two cities is not zero

In [57]:
#difference between the mean

lower=(Mean-Mean1)-2.57*(math.sqrt(var_cal/n1+ var_flo/n2))
print(lower)
upper=(Mean-Mean1)+2.57*(math.sqrt(var_cal/n1+ var_flo/n2))
print(upper)

#norm.cdf(2.5758293035489004)-norm.cdf(-2.5758293035489004)

0.006291841762153222
0.19504149157118014


0.006<=(u1-u2)<=0.195

We are 99% confident that the actual difference between the mean price of a liter of health drink between two cities is between 0.006 and 0.195

The point estimate is(Mean-Mean1)=0.10066

In [60]:
#END